src/main/java/core/basesyntax/IsoscelesTrapezoid.java

```diff
@@ -0,0 +1,29 @@
+package core.basesyntax;
+
+public class IsoscelesTrapezoid extends Figure {
+    private final double upperA; //Length of the upper base
+    private final double lowerB; // Length of the lower base
+    private final double height; // height
+
+    public IsoscelesTrapezoid(double upperA, double lowerB, double height) {
+        this.upperA = upperA;
+        this.lowerB = lowerB;
+        this.height = height;
+        setColor(new ColorSupplier().getRandomColor());
```

Comment:
Creating a new instance of ColorSupplier in the constructor is not the best practice. You should pass a Color object as a parameter to the constructor of a figure instead of generating it inside the figure classes.